src/commands/Fun/Random/roll.ts

```diff
@@ -3,47 +3,126 @@ import { CommandStore, KlasaMessage } from 'klasa';
 import { Message } from 'discord.js';
 import { oneLine } from 'common-tags';
 
+const DICE_REGEX = /(?<count>\d{1,2})?d(?<sides>\d{1,4})(?<explode>!)?(?<keep>kl?(?<keepCount>\d{1,2}))?/;
+
+type KeepType = 'highest' | 'lowest' | false;
+
+interface RollSpec {
+	input: string;
+	count: number;
+	sides: number;
+	explodes: boolean;
+	keep: KeepType;
+	keepCount?: number;
+}
+
+interface DiceResult {
+	spec: RollSpec;
+	rolls: number[];
+}
+
 export default class extends SteveCommand {
 
 	public constructor(store: CommandStore, file: string[], directory: string) {
 		super(store, file, directory, {
 			aliases: ['dice'],
 			description: 'Roll dice!',
-			examples: ['roll 1d6', 'roll 5d10!'],
+			examples: ['roll 1d6', 'roll d20', 'roll 5d10!', 'roll 1d8|4d6', 'roll 6d12k1', 'roll 6d12kl2'],
 			extendedHelp: oneLine`Use standard dice notation. You can add a "!" at the end of your roll to use exploding dice. You can roll
 				up to 10 dice with up to 1,000 sides each.`,
-			usage: '<roll:reg/(?<dice>\\d{1,2})d(?<sides>\\d{1,4})(?<explode>!)?/>',
+			usage: '<spec:dice> [...]',
 			helpUsage: '[1-9]d[1-999] (!)'
 		});
 
-		this
-			.customizeResponse('roll', 'You can roll up to 10 dice with up to 1,000 sides each.');
+		this.customizeResponse('roll', 'You can roll up to 10 dice with up to 1,000 sides each.');
+
+		this.createCustomResolver('dice', (arg): RollSpec => {
+			const match = DICE_REGEX.exec(arg);
+
+			let count = parseInt(match.groups.count) ?? 1;
```

Comment:
```suggestion
			let count = parseInt(match.groups.count, 10) ?? 1;
```

Reply:
Can you add the radixes to parseInt()? Gonna be switching up our eslint rules pretty soon and omitting the radix will make the linter angry lol.